include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -300,23 +301,24 @@ JEMALLOC_ALWAYS_INLINE void
 arena_dalloc_large(tsdn_t *tsdn, void *ptr, tcache_t *tcache, szind_t szind,
     bool slow_path) {
 	assert (!tsdn_null(tsdn) && tcache != NULL);
-	if (szind < TCACHE_NBINS_MAX &&
-	    !tcache_bin_disabled(szind, &tcache->bins[szind],
-	    tcache->tcache_slow)) {
-		if (config_prof && unlikely(szind < SC_NBINS)) {
-			arena_dalloc_promoted(tsdn, ptr, tcache, slow_path);
-		} else {
+	bool is_sampled = config_prof && szind < SC_NBINS;
```

Comment:
Looks good! Last request: let's call this `is_sample_promoted`. I forgot that large sampled isn't covered here. Also can you put this change into a separate commit, and callout the bugfix in the commit msg? It only affects when tcache_max is tuned down, but we have it in some binaries already.